### PR TITLE
Define MyAccount view for a user without local credentials

### PR DESCRIPTION
**Pull request by minolih**
### Checklist
- [x] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/myaccount/src/components/create-password/create-password.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline apps/myaccount/src/components/create-password/create-password.tsx:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/myaccount/src/components/create-password/create-password.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { AlertInterface } from "../../models";
+import { EditSection, SettingsSection } from "../shared";
+
+/**
+ * Import password strength meter dynamically.
+ */
+const PasswordMeter = React.lazy(() => import("react-password-strength-bar"));
+
```
Comment (brionmario) [inline apps/myaccount/src/components/create-password/create-password.tsx:30]:
```suggestion
```

File: apps/myaccount/src/components/create-password/create-password.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    onAlertFired: (alert: AlertInterface) => void;
+}
+
+/**
+ * Change password component.
+ *
+ * @param {CreatePasswordProps} props - Props injected to the change password component.
+ * @return {JSX.Element}
+ */
+export const CreatePassword: FunctionComponent<CreatePasswordProps> = (props: CreatePasswordProps): JSX.Element => {
```
Comment (brionmario) [inline apps/myaccount/src/components/create-password/create-password.tsx:46]:
```suggestion
export const CreatePassword: FunctionComponent<CreatePasswordProps> = (props: CreatePasswordProps): ReactElement => {
```

File: apps/myaccount/src/components/create-password/create-password.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ */
+interface CreatePasswordProps extends TestableComponentInterface {
+    onAlertFired: (alert: AlertInterface) => void;
+}
+
+/**
+ * Change password component.
+ *
+ * @param {CreatePasswordProps} props - Props injected to the change password component.
+ * @return {JSX.Element}
```
Comment (brionmario) [inline apps/myaccount/src/components/create-password/create-password.tsx:44]:
```suggestion
 *
 * @return {ReactElement}
```

File: apps/myaccount/src/components/create-password/create-password.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ */
+export const CreatePassword: FunctionComponent<CreatePasswordProps> = (props: CreatePasswordProps): JSX.Element => {
+
+    const {
+        ["data-testid"]: testId
+    } = props;
+
+    const { t } = useTranslation();
+
```
Comment (brionmario) [inline apps/myaccount/src/components/create-password/create-password.tsx:53]:
```suggestion
```

File: apps/myaccount/src/components/create-password/create-password.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    return (
+        <SettingsSection
+            data-testid={ `${testId}-settings-section` }
+            description={ t("myAccount:sections.createPassword.description") }
+            header={ t("myAccount:sections.createPassword.heading") }
+            primaryAction={ t("myAccount:sections.createPassword.actionTitles.create") }
+            primaryActionIcon="key"
+            showActionBar={ true }
+            disabled={ true }
+        >
```
Comment (brionmario) [inline apps/myaccount/src/components/create-password/create-password.tsx:64]: Lets remove `</SettingsSection>`
```suggestion
        />
```

File: apps/myaccount/src/components/overview/federated-user-overview.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline apps/myaccount/src/components/overview/federated-user-overview.tsx:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/myaccount/src/pages/overview.tsx [hunk 8]
```diff
@@ -16,10 +16,12 @@
  * under the License.
  */
 
-import React, {ReactElement, useEffect, useState} from "react";
+import { ProfileConstants } from "@wso2is/core/constants";
+import React, { ReactElement, useEffect, useState } from "react";
+import { Simulate } from "react-dom/test-utils";
```
Comment (brionmario) [inline apps/myaccount/src/pages/overview.tsx:21]: ```suggestion```

File: apps/myaccount/src/pages/overview.tsx [hunk 9]
```diff
@@ -35,28 +37,67 @@ const OverviewPage = (): ReactElement => {
...
+                    {
+                        t(
+                            "myAccount:pages:overview.title",
+                            { firstName: userProfileName }
+                        )
+                    }
+                    { userSource && isFederatedUser &&
+                        <div className="overview-page-header">
+                            {
+                                "(Signed up via " + userSource + ")"
```
Comment (brionmario) [inline apps/myaccount/src/pages/overview.tsx:81]: Let's add i18n strings.
```suggestion
                                "(Signed up via " + userSource + ")"
```

File: apps/myaccount/src/components/create-password/create-password.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+ * Import password strength meter dynamically.
+ */
+const PasswordMeter = React.lazy(() => import("react-password-strength-bar"));
+
+
+/**
+ * Prop types for the change password component.
+ * Also see {@link CreatePassword.defaultProps}
+ */
+interface CreatePasswordProps extends TestableComponentInterface {
```
Comment (brionmario) [inline apps/myaccount/src/components/create-password/create-password.tsx:36]:
```suggestion
interface CreatePasswordPropsInterface extends TestableComponentInterface {
```

File: apps/myaccount/src/components/overview/federated-user-overview.tsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+import { AppConstants } from "../../constants";
+import { FeatureConfigInterface } from "../../models";
+import { AppState } from "../../store";
+
+/**
+ * Overview component.
+ *
+ * @return {JSX.Element}
+ */
+export const FederatedUserOverview: FunctionComponent<{}> = (): JSX.Element => {
```
Comment (brionmario) [inline apps/myaccount/src/components/overview/federated-user-overview.tsx:33]: lets create a `type`

type FederatedUserOverviewPropsInterface = TestableComponentInterface;

```suggestion
export const FederatedUserOverview: FunctionComponent<FederatedUserOverviewPropsInterface> = (): JSX.Element => {
```

File: apps/myaccount/src/pages/overview.tsx (hunk 9 above)
Comment (minolih) [reply apps/myaccount/src/pages/overview.tsx:80]: Sure. I will not commit this as this has been relocated to a different file. There I will use i18n string.

File: apps/myaccount/src/components/federated-associations/federated-associations.tsx [hunk 12]
```diff
@@ -84,6 +97,17 @@ export const FederatedAssociations: FunctionComponent<FederatedAssociationsProps
         getFederatedAssociationsList();
     }, []);
 
+    /**
+     *
+     */
```
Comment (thivi) [inline apps/myaccount/src/components/federated-associations/federated-associations.tsx:102]: Can we add a comment here or remove this comment block?

File: apps/myaccount/src/pages/account-security.tsx [hunk 13]
```diff
@@ -16,11 +16,14 @@
...
 
+import { ProfileConstants } from "@wso2is/core/constants";
 import { hasRequiredScopes, isFeatureEnabled } from "@wso2is/core/helpers";
+import { TestableComponentInterface } from "@wso2is/core/models";
 import React, {
+    FunctionComponent,
     ReactElement,
     useEffect,
-    useRef
+    useRef, useState
```
Comment (thivi) [inline apps/myaccount/src/pages/account-security.tsx:26]:
```suggestion
    useRef, 
    useState
```

File: apps/myaccount/src/pages/overview.tsx [hunk 14]
```diff
@@ -16,51 +16,121 @@
...
+/**
+ * Default properties for the {@link OverviewPage} component.
+ * See type definitions in {@link OverviewPage}
+ */
+OverviewPage.defaultProps = {
+    enableNonLocalCredentialUserView: commonConfig.nonLocalCredentialUser.enableNonLocalCredentialUserView,
+    enableAlternateWidgetLayout: commonConfig.overviewPage.enableAlternateWidgetLayout
+};
+
+
```
Comment (thivi) [inline apps/myaccount/src/pages/overview.tsx:133]:
```suggestion
```

**Review by thivi — APPROVED** [review]

File: apps/myaccount/src/components/federated-associations/federated-associations.tsx (hunk 12 above)
Comment (minolih) [reply apps/myaccount/src/components/federated-associations/federated-associations.tsx]: A comment is added now.

**Review by thivi — APPROVED** [review]

File: apps/myaccount/src/components/federated-associations/federated-associations.tsx [hunk 15]
```diff
@@ -16,17 +16,22 @@
...
 import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
 import { Button, Grid, Icon, List, Modal, Popup } from "semantic-ui-react";
 import { deleteFederatedAssociation, getFederatedAssociations } from "../../api/federated-associations";
 import { getSettingsSectionIcons } from "../../configs";
+import { commonConfig } from "../../extensions";
 import {
     AlertInterface,
-    AlertLevels
+    AlertLevels, AuthStateInterface
```
Comment (NipuniBhagya) [inline apps/myaccount/src/components/federated-associations/federated-associations.tsx:31]:
```suggestion
    AlertLevels, 
    AuthStateInterface
```

File: apps/myaccount/src/components/overview/widgets/profile-widget.tsx [hunk 16]
```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NipuniBhagya) [inline apps/myaccount/src/components/overview/widgets/profile-widget.tsx:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/myaccount/src/pages/account-security.tsx [hunk 17]
```diff
@@ -109,8 +156,20 @@ const AccountSecurityPage = (props: RouteProps): ReactElement => {
                             </Grid.Column>
                         </Grid.Row>
                     ) : null }
-
-                { !isReadOnlyUser &&
+                { /* Create password section temporarily commented until feature is planned. */ }
```
Comment (NipuniBhagya) [inline apps/myaccount/src/pages/account-security.tsx:159]: Shall we create a git issue to track this and link it in the doc comment?

File: apps/myaccount/src/pages/overview.tsx [hunk 18]
```diff
@@ -16,51 +16,120 @@
...
 import { AuthStateInterface } from "../models";
 import { AppState } from "../store";
 
+/**
+ * Prop types for the overview page.
+ */
+interface OverviewPagePropsInterface extends TestableComponentInterface {
+    enableNonLocalCredentialUserView?: boolean;
+    enableAlternateWidgetLayout?: boolean;
+
```
Comment (NipuniBhagya) [inline apps/myaccount/src/pages/overview.tsx:37]:
```suggestion
```

File: apps/myaccount/src/pages/overview.tsx [hunk 19]
```diff
@@ -16,51 +16,120 @@
...
+    useEffect(() => {
+        if (!enableNonLocalCredentialUserView) {
+            return;
+        }
+        // Verifies if the user is a user without local credentials.
+        if (!profileDetails?.profileInfo?.[ProfileConstants.SCIM2_ENT_USER_SCHEMA]?.
+            [ProfileConstants?.SCIM2_SCHEMA_DICTIONARY.get("LOCAL_CREDENTIAL_EXISTS")]) {
+            setIsNonLocalCredentialUser(true);
+        }
+
```
Comment (NipuniBhagya) [inline apps/myaccount/src/pages/overview.tsx:80]:
```suggestion
```

File: apps/myaccount/src/pages/overview.tsx [hunk 20]
```diff
@@ -16,51 +16,120 @@
...
 
+/**
+ * Default properties for the {@link OverviewPage} component.
+ * See type definitions in {@link OverviewPage}
+ */
+OverviewPage.defaultProps = {
+    enableNonLocalCredentialUserView: commonConfig.nonLocalCredentialUser.enableNonLocalCredentialUserView,
+    enableAlternateWidgetLayout: commonConfig.overviewPage.enableAlternateWidgetLayout
+};
+
```
Comment (NipuniBhagya) [inline apps/myaccount/src/pages/overview.tsx:132]:
```suggestion
```

**Review by NipuniBhagya — CHANGES_REQUESTED** [review]

File: apps/myaccount/src/pages/account-security.tsx (hunk 17 above)
Comment (minolih) [reply apps/myaccount/src/pages/account-security.tsx:159]: I will get this issue created.

**Review by NipuniBhagya — APPROVED** [review]

**Pull request merged**